src/neural/cuda/layers.cc

```diff
@@ -1225,6 +1141,425 @@ ResidualBlock<DataType>::~ResidualBlock() {
   }
 }
 
+template <typename DataType>
+void allocAndUpload(DataType** gpu_dest, std::vector<float> cpu_src,
+                    void* scratch) {
+  size_t size = cpu_src.size() * sizeof(DataType);
+  if (size == 0) {
+    *gpu_dest = nullptr;
+    return;
+  }
+  ReportCUDAErrors(cudaMalloc(gpu_dest, size));
+  ReportCUDAErrors(
+      cudaMemcpy(scratch, &cpu_src[0], cpu_src.size() * sizeof(float), cudaMemcpyHostToDevice));
+  copyTypeConverted((DataType*)(*gpu_dest), (float*)scratch,
+                    (int)cpu_src.size(), 0);
+}
+
+template <typename DataType>
+AttentionPolicyHead<DataType>::AttentionPolicyHead(BaseLayer<DataType>* ip,
+                                                   const LegacyWeights& weights,
+                                                   void* scratch)
+    : BaseLayer<DataType>(ip->GetC(), 8, 8, ip) {
```

Comment:
ip->GetC() isn't correct here ... its a bit ugly - but technically 67 would work?

Reply:
maybe better to set H and W to 1 and instead set C to 64*64+24*8

Reply:
fixed, thanks.